src/language-handlebars/utils.js

```diff
@@ -11,6 +11,84 @@ function isGlimmerComponent(node) {
   return tagFirstChar.toUpperCase() === tagFirstChar || isLocal;
 }
 
+function isWhitespaceNode(node) {
+  return node.type === "TextNode" && !/\S/.test(node.chars);
+}
+
+function isNodeOfSomeType(node, types) {
+  return node && types.some(type => node.type === type);
+}
+
+function isParentOfSomeType(path, types) {
+  const parentNode = path.getParentNode(0);
+  return isNodeOfSomeType(parentNode, types);
+}
+
+function isPreviousNodeOfSomeType(path, types) {
+  const previousNode = getPreviousNode(path);
+  return isNodeOfSomeType(previousNode, types);
+}
+
+function isNextNodeOfSomeType(path, types) {
+  const nextNode = getNextNode(path);
+  return isNodeOfSomeType(nextNode, types);
+}
+
+function getPreviousNode(path, lookBack = 1) {
+  const node = path.getValue();
+  const parentNode = path.getParentNode(0);
+
+  const children = parentNode && (parentNode.children || parentNode.body);
+  if (children) {
+    const nodeIndex = children.indexOf(node);
+    if (nodeIndex > 0) {
+      const previousNode = children[nodeIndex - lookBack];
+      return previousNode;
+    }
+  }
+}
```

Comment:
```suggestion
function getPreviousNode(path, lookBack = 1) {
  const node = path.getValue();
  const parentNode = path.getParentNode(0) || {};
  const children = parentNode.children || parentNode.body || [];
  const index = children.indexOf(node)
  return index !== -1 && children[index - lookBack];
}
```

Reply:
I have applied your suggestions. It unlocks a few good improvements. Do you mind having a look at my last commit?